Make `cilium status` output more succinct by default

joestringer opened the pull request:
```release-note-minor
Simplify output of `cilium status` by default, add new `--verbose`, `--brief` options
```

Add `--verbose` mode and some extra options to choose how much output you want from the cilium status API. Gather the verbose output in bugtool. Fold in a minor change to phrase controller states in terms of success rather than failure.

Example output with this series:

```
$ cilium status                                                                                                                                                                                                                                                
KVStore:                    Ok         Consul: 172.17.0.2:8300                  
ContainerRuntime:           Ok                                                  
Kubernetes:                 Disabled                                            
Cilium:                     Ok         OK                                       
NodeMonitor:                Disabled                                            
Cilium health daemon:       Warning   Get http:///var/run/cilium/health.sock/v1beta/hello: dial unix /var/run/cilium/health.sock:
 connect: no such file or directory                                             
IPv4 address pool:   3/65535 allocated                                                   
IPv6 address pool:   2/65535 allocated                                                   
Controller Status:     3/3 healthy
```

There's also this command for a brief, one-line summary:
```
$ cilium status --brief
OK
```

joestringer committed:
cilium: Simplify output for 'cilium status'

Introduce three new arguments to 'cilium status':

* allAddresses: Print all addr allocations (default: print only the count)
* allNodes: Print all known nodes (default: only print localhost info)
* verbose: Equivalent to --all-addresses --all-controllers --all-nodes

This ensures that by default in large environments, 'cilium status' will
not flood the terminal of the person trying to get information, but
still allows more information to be gathered on demand.

Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
bugtool: Capture verbose output of cilium status

Fixes: #2749
Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
controller: Be more optimistic about states

Frame controller output in terms of how many are OK, rather than how
many are failing.

Signed-off-by: Joe Stringer <joe@covalent.io>

tgraf reviewed (dismissed):
Can you also add a `--brief` option which prints the overall status on a single line? We need this for the Kubernetse healthcheck.

joestringer committed:
pkg/ip: Add helper to count IPs in subnet

This will be used by an upcoming commit to report the max number of IPs
that may be allocated by cilium.

Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
cilium: Report max IP pool in 'cilium status'

Signed-off-by: Joe Stringer <joe@covalent.io>

joestringer committed:
cilium: Add cilium status --brief

The new `--brief` argument prints a single line summary of the status
which prints either "OK" or "error: ...", which prints only one of the
errors currently reported in the system, with preference in this order:

* Cilium daemon
* Container Runtime
* Kvstore
* Kubernetes
* Cilium-health daemon
* Controllers inside Cilium daemon

Signed-off-by: Joe Stringer <joe@covalent.io>
